### DISCUSSION TITLE
Bevy Render Rework: Postprocessing Effects

### DISCUSSION THREAD

**Discussion by james7132** [bevyengine/bevy#2454] [Ideas]
With the renderer rework slowly reaching production quality, it may be a good time to start looking at the standard gambit of postprocessing effects commonly seen across game engines. This initial post is to gather input/ideas/decisions. Here's an initial list of potential effects that might be useful, (* marks effects that will be required for a HDR workflow):

 - Bloom*
 - Color correction*
 - Tonemapping (ACES?)*
 - Depth of Field
 - Chromatic Aberration
 - White Balance
 - Lens Distortion
 - Lens Flare
 - Film Grain
 - Motion Blur
 - Panini Projection
 - Vignette
 - Antialiasing (FXAA, SMAA, TXAA)

### Important Points to Consider
 - Ordering between each effect is important, Unity's Postprocessing notes that using DoF after Bloom adds notable artifacts
 - Merging multiple similar effects into one shader as a ubereffect might be a good idea.
 - Should we use a separate pass from this or just run it as the end of the main pass?

**Comment by mtsr** [comment]
I'd like to add a Composite step as required for Weighted Blended Order Independent Transparency (http://jcgt.org/published/0002/02/09/)

And I'm interested in the API. I did a pretty rough first implementation for the old renderer with [FullscreenPassNode](http://jcgt.org/published/0002/02/09/), but the new renderer is pretty different.
